pkg/cluster/denyassignment.go

```diff
@@ -17,9 +18,18 @@ func (m *manager) createOrUpdateDenyAssignment(ctx context.Context) error {
 	}
 
 	// needed for AdminUpdate so it would not block other steps
-	if m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.SPObjectID == "" {
-		m.log.Print("skipping createOrUpdateDenyAssignment: SPObjectID is empty")
-		return nil
+	if m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		for _, i := range m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities {
+			if i.ObjectID == "" {
+				m.log.Print(fmt.Sprintf("skipping createOrUpdateDenyAssignment: ObjectID for identity %s is empty", i.OperatorName))
+				return nil
+			}
+		}
+	} else {
+		if m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.SPObjectID == "" {
```

Comment:
Will there be a risk of nil pointer dereference here? (when both PlatformWorkloadIdentityProfile and ServicePrincipalProfile are nil?)

Reply:
There _shouldn't_ be, static validation should prevent a scenario where both PlatformWorkloadIdentityProfile and ServicePrincipalProfile are nil https://github.com/Azure/ARO-RP/blob/master/pkg/api/v20240812preview/openshiftcluster_validatestatic.go#L488-L490

Would you prefer to explicitly check for a nil pointer first?

Reply:
I added a check (with associated unit test) to prevent a nil pointer dereference in this case if the static validation doesn't catch this scenario